packages/react-next/src/components/Checkbox/Checkbox.base.tsx

```diff
@@ -0,0 +1,167 @@
+import { KeytipData } from '../../KeytipData';
+import { useId, useControllableValue } from '@uifabric/react-hooks';
+
+const getClassNames = classNamesFunction<ICheckboxStyleProps, ICheckboxStyles>();
+
+export const CheckboxBase = React.memo(
+  React.forwardRef((props: ICheckboxProps, ref: React.Ref<HTMLDivElement>) => {
+    const {
+      className,
+      disabled,
+      inputProps,
+      name,
+      boxSide = 'start',
+      theme,
+      ariaLabel,
+      ariaLabelledBy,
+      ariaDescribedBy,
+      styles,
+      checkmarkIconProps,
+      ariaPositionInSet,
+      ariaSetSize,
+      keytipProps,
+      title,
+      label,
+      onChange,
+    } = props;
+
+    const checkBox = React.useRef<HTMLInputElement>(null);
+    const [isChecked, setIsChecked] = useControllableValue(props.checked, props.defaultChecked, onChange);
+    const [isIndeterminate, setIsIndeterminate] = useControllableValue(props.indeterminate, props.defaultIndeterminate);
+
+    useDebugWarning(props);
+    useComponentRef(props, isChecked, isIndeterminate, checkBox);
+
+    const id = useId('checkbox-', props.id);
+    const classNames: { [key in keyof ICheckboxStyles]: string } = getClassNames(styles!, {
+      theme: theme!,
+      className,
+      disabled,
+      indeterminate: isIndeterminate,
+      checked: isChecked,
+      reversed: boxSide !== 'start',
+      isUsingCustomLabelRender: !!props.onRenderLabel,
+    });
+
+    const onRenderLabel = (): JSX.Element | null => {
+      return label ? (
+        <span aria-hidden="true" className={classNames.text} title={title}>
+          {label}
+        </span>
+      ) : null;
+    };
+
+    //#region Input callbacks
+    const _onFocus = React.useCallback(
+      (ev: React.FocusEvent<HTMLElement>): void => {
+        if (inputProps?.onFocus) {
+          inputProps.onFocus(ev);
+        }
+      },
+      [inputProps?.onFocus],
+    );
+
+    const _onBlur = React.useCallback(
+      (ev: React.FocusEvent<HTMLElement>): void => {
+        if (inputProps?.onBlur) {
+          inputProps.onBlur(ev);
+        }
+      },
+      [inputProps?.onBlur],
+    );
+
+    const _onChange = React.useCallback(
+      (ev: React.ChangeEvent<HTMLElement>): void => {
+        if (!isIndeterminate) {
+          setIsChecked(!isChecked, ev);
+        } else {
+          // If indeterminate, clicking the checkbox *only* removes the indeterminate state (or if
+          // controlled, lets the consumer know to change it by calling onChange). It doesn't
+          // change the checked state.
+          setIsChecked(!!isChecked, ev);
+          setIsIndeterminate(false);
+        }
+      },
+      [isChecked, isIndeterminate],
+    );
+    //#endregion
+
+    return (
+      <KeytipData keytipProps={keytipProps} disabled={disabled}>
+        {(keytipAttributes: any): JSX.Element => (
+          <div className={classNames.root} title={title} ref={ref}>
+            <FocusRects />
```

Comment:
use `useFocusRects` instead. `<FocusRects />` is just a wrapper component which is meant for be used only in class components where hook cannot be called directly